Datagrid/ProxyQuery.php

```diff
@@ -355,4 +347,18 @@ protected function getFixedQueryBuilder(QueryBuilder $queryBuilder)
 
         return $queryBuilder;
     }
+
+    private function addOrderedColumns(QueryBuilder $queryBuilder)
+    {
+        /* For each ORDER BY clause defined directly in the DQL parts of the query,
+           we add an entry in the SELECT clause. */
+        $dqlParts = $queryBuilder->getDqlParts();
+        if ($dqlParts['orderBy'] && count($dqlParts['orderBy'])) {
+            foreach ($dqlParts['orderBy'] as $part) {
```

Comment:
Could probably be simplified with a cast : https://3v4l.org/D6cRP

```php
<?php
$dqlParts = $queryBuilder->getDqlParts();
foreach ((array) $dqlParts['orderBy'] as $part) {
```

Reply:
I like it, thanks.